src/content-helper/dashboard-page/pages/traffic-boost/preview/preview.tsx

```diff
@@ -0,0 +1,406 @@
+
+	const [ activeLink, setActiveLink ] = useState<TrafficBoostLink>( providedActiveLink );
+	const [ activePost, setActivePost ] = useState<HydratedPost>( providedActiveLink.targetPost );
+
+	const [ selectedText, setSelectedText ] = useState<TextSelection | null>( null );
+	const [ previewUrl, setPreviewUrl ] = useState<string>( '' );
+	const [ totalItems, setTotalItems ] = useState<number>( 0 );
+	const [ itemIndex, setItemIndex ] = useState<number>( 0 );
+
+	const {
+		createSuccessNotice,
+	} = useDispatch( 'core/notices' );
+
+	const {
+		post,
+		suggestions,
+		inboundLinks,
+	} = useSelect( ( select ) => {
+		return {
+			post: select( TrafficBoostStore ).getCurrentPost(),
+			suggestions: select( TrafficBoostStore ).getSuggestions(),
+			inboundLinks: select( TrafficBoostStore ).getInboundLinks(),
+		};
+	}, [ ] );
+
+	const {
+		setSelectedLink,
+		removeSuggestion,
+		removeInboundLink,
+		addInboundLink,
+		setSelectedTab,
+		setIsAccepting,
+		setIsRemoving,
+	} = useDispatch( TrafficBoostStore );
+
+	/**
+	 * Sets the active link to the provided active link.
+	 *
+	 * @since 3.18.0
+	 */
+	useEffect( () => {
+		setActiveLink( providedActiveLink );
+	}, [ providedActiveLink ] );
+
+	/**
+	 * Sets the active post to the target post of the active link,
+	 * and unsets the text selection when the active link changes.
+	 *
+	 * @since 3.18.0
+	 */
+	useEffect( () => {
+		setActivePost( activeLink.targetPost );
+		setIsInboundLink( ! activeLink.isSuggestion );
+		setSelectedText( null );
+	}, [ activeLink ] );
+
+	/**
+	 * Sets the total items and item index based on the active link.
+	 *
+	 * @since 3.18.0
+	 */
+	useEffect( () => {
+		if ( activeLink.isSuggestion ) {
+			setTotalItems( suggestions?.length ?? 0 );
+			setItemIndex( suggestions?.indexOf( activeLink ) + 1 );
+		} else {
+			setTotalItems( inboundLinks?.length ?? 0 );
+			setItemIndex( inboundLinks?.indexOf( activeLink ) + 1 );
+		}
+	}, [ activeLink, inboundLinks, suggestions ] );
+
+	/**
+	 * Sets the preview URL based on the active post and frontend preview setting.
+	 *
+	 * @since 3.18.0
+	 */
+	useEffect( () => {
+		if ( ! activePost ) {
+			return;
+		}
+
+		const newUrl = ! isFrontendPreview
+			? addQueryArgs( `${ window.location.origin }/wp-admin/admin-ajax.php`, {
+				action: 'parsely_post_preview',
+				post_id: activePost.id,
+				_wpnonce: window._parsely_traffic_boost_preview_nonce ?? '',
+			} )
+			: addQueryArgs( activePost.guid.raw, {
+				parsely_preview: 'true',
+				_wpnonce: window._parsely_traffic_boost_preview_nonce ?? '',
+			} );
+
+		// Only set loading state if URL actually changes.
+		if ( newUrl !== previewUrl ) {
+			setIsLoading( true );
+			setPreviewUrl( newUrl );
+		}
+	}, [ activePost, isFrontendPreview, previewUrl ] );
+
+	/**
+	 * Opens the post in a new tab.
+	 *
+	 * @since 3.18.0
+	 */
+	const openPostInNewTab = () => {
+		if ( ! activePost?.guid?.raw ) {
+			return;
+		}
+
+		window.open( activePost.guid.raw, '_blank' );
```

Comment:
In the functions here that use `window.open`, do we need/want to provide a `noopener` param?